includes/class-wc-stripe-customer.php

```diff
@@ -193,6 +193,34 @@ protected function generate_customer_request( $args = [] ) {
 		return wp_parse_args( $args, $defaults );
 	}
 
+	/**
+	 * If customer does not exist, create a new customer. Else retrieve the Stripe customer through the API to check it's existence.
+	 * Recreate the customer if it does not exist in this Stripe account.
+	 *
+	 * @return string Customer ID
+	 *
+	 * @throws WC_Stripe_Exception
+	 */
+	public function maybe_create_customer( $should_recreate_customer = false ) {
```

Comment:
The [attach_source](https://github.com/woocommerce/woocommerce-gateway-stripe/blob/f4fd6119a58c83c2aa26ded8f362df3b52356150/includes/class-wc-stripe-customer.php#L405) function used to [recreate a customer](https://github.com/woocommerce/woocommerce-gateway-stripe/blob/f4fd6119a58c83c2aa26ded8f362df3b52356150/includes/class-wc-stripe-customer.php#L405) apart from sending the `/attach` customer request. We want to avoid the `/attach` customer request but in a case when the customer does not exist yet, we need the recreate customer part. This function is introduced to handle that scenario.